tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/MessageDrivenEJBTest.java

```diff
@@ -0,0 +1,225 @@
+        new ResourceVerifier(EJBFixtures.messageDrivenEJBAddress(MESSAGE_DRIVEN_EJB_DEPLOYMENT_EXECUTION_TIME_NAME,
+            MessageDrivenEJBExecutionTime.class),
+            client)
+            .verifyExists()
+            .verifyAttribute("execution-time", Long.valueOf(actualExecutionTime));
+    }
+
+    @Test
+    public void verifyInvocations() throws Exception {
+        int numberOfInvocations = Random.number(1, 5);
+        invoke(numberOfInvocations, MESSAGE_DRIVEN_EJB_INVOCATIONS_DEPLOYMENT);
+        console.navigate(MESSAGE_DRIVEN_EJB_INVOCATIONS_DEPLOYMENT.getPlaceRequest());
+        Assert.assertEquals(
+            "Number of invocations in the form should be equal to the total number of invocations executed",
+            numberOfInvocations,
+            page.getMessageDrivenBeanEJBForm().intValue("invocations"));
+        new ResourceVerifier(
+            EJBFixtures.messageDrivenEJBAddress(MESSAGE_DRIVEN_EJB_INVOCATIONS_DEPLOYMENT_NAME,
+                MessageDrivenEJBInvocations.class),
+            client)
+            .verifyExists()
+            .verifyAttribute("invocations", Long.valueOf(numberOfInvocations));
+    }
+
+    @Test
+    public void verifyPeakConcurrentInvocations() throws Exception {
+        int numberOfConcurrentInvocations = Random.number(2, 5);
+        Runnable invocationTask = () -> invoke(1, MESSAGE_DRIVEN_EJB_PEAK_CONCURRENT_INVOCATIONS_DEPLOYMENT);
+        for (int i = 0; i < numberOfConcurrentInvocations; ++i) {
+            new Thread(invocationTask).start();
+        }
+        console.navigate(MESSAGE_DRIVEN_EJB_PEAK_CONCURRENT_INVOCATIONS_DEPLOYMENT.getPlaceRequest());
+        Assert.assertEquals("Number of peak concurrent invocations should be correct", numberOfConcurrentInvocations,
+            page.getMessageDrivenBeanEJBForm().intValue("peak-concurrent-invocations"));
+        new ResourceVerifier(
+            EJBFixtures.messageDrivenEJBAddress(MESSAGE_DRIVEN_EJB_PEAK_CONCURRENT_INVOCATIONS_DEPLOYMENT_NAME,
+                MessageDrivenEJBPeakConcurrentInvocations.class), client)
+            .verifyExists()
+            .verifyAttribute("peak-concurrent-invocations", Long.valueOf(numberOfConcurrentInvocations));
+    }
+
+    @Test
+    public void verifyRunAsRole() throws Exception {
+        console.navigate(MESSAGE_DRIVEN_EJB_WITH_ROLES_DEPLOYMENT.getPlaceRequest());
+        Assert.assertEquals("Declared roles in the form should be equal to the EJB's run as role", EJBFixtures.ROLE_1,
+            page.getMessageDrivenBeanEJBForm().value("run-as-role"));
+        new ResourceVerifier(
+            EJBFixtures.messageDrivenEJBAddress(MESSAGE_DRIVEN_EJB_WITH_ROLES_DEPLOYMENT_NAME,
+                MessageDrivenEJBWithRoles.class),
+            client)
+            .verifyExists()
+            .verifyAttribute("run-as-role", EJBFixtures.ROLE_1);
+    }
+
+    private static void invoke(int numberOfInvocations, EJBDeployment ejbDeployment) {
```

Comment:
Seems similar method is written more times. What about moving this method to some library class? WDYT?